### ISSUE TITLE
Selective cell tracking during manual editing of a segmented movie

### ISSUE THREAD

**Issue by faultby** [SchmollerLab/Cell_ACDC#264]
### Description:

Hi!

First of all, I want to say that CellACDC is a great application and it helps us with our project! 

I am currently working with segmented movies. Naturally, they require manual correction for segmentation, as well as editing cell IDs during real-time tracking in the GUI. The movies I am working with are of long experiments, during which cells undergo multiple rounds of cell division. They usually beginn with just 2-8 cells in the field of view and, by the end, the cell count gets up to a hundred. For now, I am interested in segmenting and tracking only cells that were present there in the first frame (or so). Currently, one would need to select those cells in each new frame using "Select objects to keep" button, or use it at the very end of the editing process. These methods work fine, however they make editing more time consuming or computationally challenging.

Therefore, it would be useful if there was a way to set up number of cells (or their IDs) that I want to keep track of while going trough frames of a movie in the "segmentation and tracking" mode of the GUI. 

Let me know if you need more info and what you think of the suggestion.

Cheers!
Alex

**Comment by dimethylglymaxim** [comment]
I usually go through the whole movie in "Segmentation and tracking" mode and only then use "Select objects to keep" to delete every "wrong" ID across the entire time-lapse:)

**Comment by ElpadoCan** [comment]
Hi Alex, 

I'm very happy to hear Cell-ACDC is actually helpful, thanks to the users' suggestions is getting better by the day, so huge thanks!

Your suggestion is great, but I think we had this discussion with Max already in issue #127. Is that similar to what you had in mind? 

Of course, I can implement what you are asking but we need to discuss how to implement it because I am not sure what is the best way. The problem is that at the beginning of a correction, you still don't know whether tracking is correct in the future. This could be a problem because if you say for example "keep cell IDs 1, 2, and 3" but then in the future cell ID 1 is assigned to the wrong cell (tracking error) then you are going to delete the wrong cell. 

It's a bit of a "chicken-egg" problem because you want to track only a subset of cells, but to know exactly where those cells are you need correct tracking which means you would not even need to check if the tracking is correct :D 

In any case, if you are fine with deleting the wrong cell every now and then (depending on how good automatic tracking was in the first place) you can already achieve that as Max suggested, or even easier, you just visit the second frame, go back to first one and in that frame select the cells to keep. Cell-ACDC will ask you to propagate this change to the future frames and more importantly, it will allow you to apply it to all UNVISITED future frames. 

Unfortunately, I don't see a perfect solution to this problem, but if you have other ideas in mind feel free to share them! 

I know it is not the answer you wanted, but I hope it is still useful somehow. Let me know what you think, thanks! 

Cheers,
Francesco

**Comment by faultby** [comment]
Thank you for your very elaborate answer!

It is indeed seem like the same idea has come to several people, as Max also essentially described it :D. I can also clearly see the problem of deleting cell masks before making sure the cells have been assigned the correct ID. Below are some thoughts on how to potentially improve the analysis:

1)  One solution would be to skip the automatic tracking all together and implement a simple manual one. In this case each frame of a movie would contain masks of cells without IDs. The user then would be able to add IDs manually, by simply clicking on desired cells in order, e.g. first clicked cell: ID = 1; second-clicked: ID = 2;... To simplify the analysis even further, one could introduce a "ghost" trace of the cells in each new frame, to keep track of where the cells were previously (similar to missing cells' "ghosts"). This solution may work well when there are "not so many" cells to include.

2) Another idea is somewhat a common-ground solution for the "chicken vs egg" problem you mentioned. Instead of deleting the masks, the feature will "hide" all the undesired IDs (and their masks). When going to the next frame and observing a tracking error, the user would be able to click on the right cell and invoke its hidden mask to be shown again. The right ID can be assigned using the existing "Edit ID" button, and the wrong cell can instead be hidden (on-demand or automatically). Once the tracking is corrected in this way, the user can delete the other masks all-together and proceed with correcting the masks. This will simplify manual tracking editing, as the main problem with it right now is essentially how much human focus it takes. By the middle of a movie it gets hard to navigate a space taken up by tens and tens of cells that change their mask colour/ID. It seems that simplifying this space by visualising only several IDs may help a ton! On top of that, introducing the "ghosts" for the "shown" IDs between frames may also help (as discussed in the first suggestion).

I hope this is not too wordy. Let me know what you think of this!

Best
Alex

**Comment by ElpadoCan** [comment]
Hi Alex,

Thank you very much for the suggestions very useful! 

Regarding point number 1, this is actually what another user requested in issue #253, so I will definitely implement it soon. 

Your point number 2 is also very interesting and that was also something I thought about. It's actually similar to the deletion ROI that we already have. Those ROIs don't permanently delete an ID but it just hides it. 

In general, I fully agree with you, it would be nice to have a way to track a subset of all the cells, especially when they become too many.

I think your points are very valid and I will definitely implement both of them. If you don't mind I will separate them into two issues. One is already the #253 issue, and the other one I just opened now, it's issue #267

I will keep you updated and thank you very much for the suggestions! 

Cheers,
Francesco

**Comment by faultby** [comment]
That sounds perfect, thank you very much! I'll keep an eye on both of them.

**Comment by ElpadoCan** [comment]
Hi Alex,

I just released `v1.2.6` with a new function called Manual tracking. There is now a new button on the top toolbar, with [this](https://raw.githubusercontent.com/SchmollerLab/Cell_ACDC/main/cellacdc/resources/icons/manual_tracking.svg) icon. Activate that button and you get a new toolbar. In the new toolbar, you can select which ID you want to track (you can also type it directly on the keyboard) and you will then be able to simply click on a cell to assign that ID. I hope it is as close as possible to what you would have expected.

To update, activate the `acdc` environment and run the command `pip install --upgrade cellacdc`.

Let me know if it works, thanks! 

Cheers,
Francesco
